### DISCUSSION TITLE
How to center a window when showing it | C++

### DISCUSSION THREAD

**Discussion by Evicia** [slint-ui/slint#8388] [Q&A] [answered]
I'm trying to center a window when showing it, 
I get the window size and then calculate the window position coordinates based on the screen size.
```
auto CenterPosition(slint::Window &window) const -> void
{
    const int screen_width = GetSystemMetrics(SM_CXSCREEN);
    const int screen_height = GetSystemMetrics(SM_CYSCREEN);
    auto size = window.size();  // 800*600
    slint::PhysicalPosition pos;
    pos.x = (screen_width - size.width) / 2;
    pos.y = (screen_height - size.height) / 2;
    window.set_position(pos);
}
```
But on Windows, calling size() on a window before it's displayed always returns 800*600, not its actual size. Calling size() on the window after it's displayed and centering it doesn't look pretty.

Is there any other way to center a window?

I will be grateful for help.

P.S I apologize for my auto-English.

**Comment by Abrahamiaan** [comment, marked as answer]
I am not sure but maybe you can just do something like

window.set_size({800, 600}) // if you want it to have constant size, just change to its actual size
CenterPosition(window);
window.show();


Or

window.show();
std::thread([&window]() {
    std::this_thread::sleep_for(std::chrono::milliseconds(50)); // Just waiting the window to be opened
    CenterPosition(window); // then only center it 
}).detach();

Ping me if you need any help

**Reply by Evicia** [reply]
I see what you mean, thank you very much.

**Reply by Abrahamiaan** [reply]
If it works, please mark it as an answer. Thanks
